pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -408,35 +420,118 @@ func fromRecipePropertiesClassificationDatamodel(e datamodel.EnvironmentRecipePr
 	return nil
 }
 
-func toRecipeConfigTerraformProvidersDatamodel(config *RecipeConfigProperties) map[string][]datamodel.ProviderConfigProperties {
+func toRecipeConfigTerraformProvidersDatamodel(config *RecipeConfigProperties) (map[string][]datamodel.ProviderConfigProperties, error) {
 	if config.Terraform == nil || config.Terraform.Providers == nil {
-		return nil
+		return nil, nil
 	}
 
 	dm := map[string][]datamodel.ProviderConfigProperties{}
 	for k, v := range config.Terraform.Providers {
 		dm[k] = []datamodel.ProviderConfigProperties{}
 
 		for _, providerAdditionalProperties := range v {
+			var addnlProperties map[string]any
+			var convertedSecrets map[string]datamodel.SecretReference
+
+			convertedSecrets = getSecretsFromProviderProperties(providerAdditionalProperties.Secrets)
+			addnlProperties = providerAdditionalProperties.AdditionalProperties
+
+			// Account for possibility of nested additional properties during unmarshalling
+			// https://json-schema.org/understanding-json-schema/reference/object#additionalproperties
+			if tmpAddnlProperties, ok := addnlProperties["additionalProperties"]; ok && addnlProperties["additionalProperties"] != nil {
+				if tmpMap, ok := tmpAddnlProperties.(map[string]any); ok {
+					addnlProperties = tmpMap
+				} else {
+					return nil, v1.ErrInvalidModelConversion
+				}
+			}
+
+			// check if additional properties contain Secrets
+			// get AdditionalProperties from providerAdditionalProperties excluding the 'secrets' key.
+			for key, value := range addnlProperties {
+				if key == "secrets" {
```

Comment:
Why are we fetching it from [additionalProperties](https://github.com/radius-project/radius/blob/3cd05edd9140bdc33a6487cf8516f117f8c6a354/pkg/corerp/api/v20231001preview/zz_generated_models.go#L1255) instead of this [explicit field assigned for secrets](https://github.com/radius-project/radius/blob/3cd05edd9140bdc33a6487cf8516f117f8c6a354/pkg/corerp/api/v20231001preview/zz_generated_models.go#L1259)?

Reply:
This is not needed and has been removed.